### PR TITLE
[Storage] [STG 96] Merge STG 96 into Main

### PR DESCRIPTION
**Pull request by weirongw23-msft**
STG 96 Beta Release

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py [hunk 1]
```diff
@@ -210,10 +210,10 @@ def __init__(
 
     @classmethod
     def from_file_url(
-            cls, file_url: str,
-            snapshot: Optional[Union[str, Dict[str, Any]]] = None,
-            credential: Optional[Union[str, Dict[str, str], "AzureNamedKeyCredential", "AzureSasCredential", "AsyncTokenCredential"]] = None,  # pylint: disable=line-too-long
-            **kwargs: Any
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py:216]: These were not merge glitches-- they in fact are too indented in `main`

**Review by vincenttran-msft — APPROVED** [review]

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py (hunk 1 above)
Comment (weirongw23-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py:216]: Nice catch! Looks like this didn't affect other files so we're good to go :P

**Pull request merged**